tests/test_wellstatefullyimplicitblackoil.cpp

```diff
@@ -114,16 +117,32 @@ struct Setup
 
 namespace {
     Opm::WellStateFullyImplicitBlackoil
-    buildWellState(const Setup& setup, const std::size_t timeStep)
+    buildWellState(const Setup& setup, const std::size_t timeStep,
+                   std::vector<Opm::ParallelWellInfo>& pinfos)
     {
         auto state  = Opm::WellStateFullyImplicitBlackoil{};
 
         const auto cpress =
             std::vector<double>(setup.grid.c_grid()->number_of_cells,
                                 100.0*Opm::unit::barsa);
 
+        auto wells = setup.sched.getWells(timeStep);
+        pinfos.resize(wells.size());
+        std::vector<Opm::ParallelWellInfo*> ppinfos(wells.size());
+        auto pw = pinfos.begin();
+        auto ppw = ppinfos.begin();
+
+        for (const auto& well : wells)
+        {
+            *pw = {well.name()};
+            *ppw = &(*pw);
+            pw->communicateFirstPerforation(true);
+            ++pw;
+            ++ppw;
+        }
+
         state.init(cpress, setup.sched,
-                   setup.sched.getWells(timeStep),
+                   wells, ppinfos,
                    timeStep, nullptr, setup.pu, setup.well_perf_data, setup.st, setup.sched.getWells(timeStep).size());
```

Comment:
I think we can skip the second call to `getWells()` here and just use `wells.size()` instead.